scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,95 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  sealed abstract case class FramingConfig private (
```

Comment:
I think benefits here are nice default to `toString` and `equals` plus `hashCode`. I am not sure why you would prevent `unapplay` to prevent pattern matching ? (from what can be seen here `https://gist.github.com/tpolecat/a5cb0dc9adeacc93f846835ed21c92d2, unapplay still works`)

Reply:
Ah, cool, I forgot about equality, that makes sense. I didn't want to prevent pattern matching, I was just curious if it worked (happy to see that it does). Thanks for the link!